distribution/registry.go

```diff
@@ -102,11 +102,7 @@ func NewV2Repository(ctx context.Context, repoInfo *registry.RepositoryInfo, end
 		scope := auth.RepositoryScope{
 			Repository: repoName,
 			Actions:    actions,
-		}
-
-		// Keep image repositories blank for scope compatibility
-		if repoInfo.Class != "image" {
-			scope.Class = repoInfo.Class
```

Comment:
@dmcgowan: I remember some discussion in an earlier PR about whether this was necessary, and as I recall you said you preferred only to set `Class` for non-images. Are you okay with changing this now?